### PR TITLE
Remove gradle from project, use Maven instead

### PR DESCRIPTION
**Pull request by frenchfrywpepper**
There should be no real functional changes to the project, just
the build mechanism is now maven instead of Gradle.

The one important change to note is that Version.java is now
removed. lightstep-tracer-java and lightstep-tracer-android
should each define their own Version.java. It makes more sense
for those to each maintain a version now that the version
numbers will not always be in sync with common.

### CODE REVIEW

File: inc-version.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+
+# To increment the version and publish a build, use the Makefile
+# make publish
+# which will call out to this script
+
+
+# Use maven-help-plugin to get the current project.version
+CURRENT_VERSION=`mvn org.apache.maven.plugins:maven-help-plugin:2.1.1:evaluate -Dexpression=project.version | grep -v '\['`
```
Comment (danielamiao) [inline inc-version.sh]: woohoo maven help plugin!

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    <scm>
+        <connection>https://github.com/lightstep/lightstep-tracer-java-common</connection>
+        <developerConnection>https://github.com/lightstep/lightstep-tracer-java-common</developerConnection>
+        <url>https://github.com/lightstep/lightstep-tracer-java-common</url>
+    </scm>
+
+    <properties>
+        <maven.compiler.source>1.7</maven.compiler.source>
+        <maven.compiler.target>1.7</maven.compiler.target>
```
Comment (danielamiao) [inline pom.xml]: I always had questions about this - I thought common was supposed to be on JRE6?

File: src/test/java/com/lightstep/tracer/shared/OptionsTest.java [hunk 3]
```diff
@@ -117,10 +117,6 @@ public void testOptionsBuilder_noComponentName() throws Exception {
         Object componentName = options.tags.get(COMPONENT_NAME_KEY);
         assertNotNull(componentName);
 
-        // the actual value will vary depending on how the tests are run, but should always
-        // contain 'Main'
-        assertTrue("Unexpected sun.java.command value", componentName.toString().contains("Main"));
```
Comment (danielamiao) [inline src/test/java/com/lightstep/tracer/shared/OptionsTest.java]: why delete this?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,60 +1,60 @@
 <a name="Pending Release"></a>
-## [Pending Release](https://github.com/lightstep/lightstep-tracer-java/compare/master...0.12.5)
+## [Pending Release](https://github.com/lightstep/lightstep-com.lightstep.tracer-java/compare/master...0.12.5)
```
Comment (frenchfrywpepper) [inline CHANGELOG.md]: I thought we should maintain the changelog history as we maintained the file history, but maybe it makes sense to start fresh for this repo alone.

File: Makefile [hunk 5]
```diff
@@ -1,44 +1,28 @@
-.PHONY: build publish pre-publish ci_test clean test inc-version
+.PHONY: build publish ci_test clean test
 
-build:
-	./gradlew build
+build: test
```
Comment (frenchfrywpepper) [inline Makefile]: Shouldn't we ensure that tests pass before allowing a build?

File: CHANGELOG.md (hunk 4 above)
Comment (danielamiao) [inline CHANGELOG.md]: oh I just meant the link is linking to `lightstep-tracer-java` without the `common` at the end

File: Makefile (hunk 5 above)
Comment (danielamiao) [inline Makefile]: 👍

File: src/test/java/com/lightstep/tracer/shared/OptionsTest.java (hunk 3 above)
Comment (frenchfrywpepper) [inline src/test/java/com/lightstep/tracer/shared/OptionsTest.java]: This was a not very valuable assertion, as stated, the actual value would depend on how the tests are run. Now that we're running them with Maven, this assertion is failing and I was unable to find a common string to look for in the value.

File: README.md [hunk 6]
```diff
@@ -1,10 +1,14 @@
-# lightstep-tracer-java-common
+# lightstep-com.lightstep.tracer-java-common
 
-[ ![Download](https://api.bintray.com/packages/lightstep/maven/lightstep-tracer-android/images/download.svg) ](https://bintray.com/lightstep/maven/) [![Circle CI](https://circleci.com/gh/lightstep/lightstep-tracer-java.svg?style=shield)](https://circleci.com/gh/lightstep/lightstep-tracer-java) [![MIT license](http://img.shields.io/badge/license-MIT-blue.svg)](http://opensource.org/licenses/MIT)
+[ ![Download](https://api.bintray.com/packages/lightstep/maven/lightstep-com.lightstep.tracer-android/images/download.svg) ](https://bintray.com/lightstep/maven/) [![Circle CI](https://circleci.com/gh/lightstep/lightstep-com.lightstep.tracer-java.svg?style=shield)](https://circleci.com/gh/lightstep/lightstep-com.lightstep.tracer-java) [![MIT license](http://img.shields.io/badge/license-MIT-blue.svg)](http://opensource.org/licenses/MIT)
```
Comment (frenchfrywpepper) [inline README.md]: done

File: CHANGELOG.md (hunk 4 above)
Comment (frenchfrywpepper) [inline CHANGELOG.md]: got it, fixed

**Comment by frenchfrywpepper** [comment]
@danielamiao For auth with bintray I put some notes in the Makefile

See https://bintray.com/lightstep for published artifacts
You must have the following entry in your settings.xml of your .m2 directory
This matches the distributionManagement/repository defined in the pom.xml

`    <server>
        <id>lightstep-bintray</id>
        <username>xxx</username>
        <password>xxx</password>
    </server>
`

File: pom.xml (hunk 2 above)
Comment (frenchfrywpepper) [inline pom.xml]: Apparently it supports 7, with a touch of 8 if you want. https://developer.android.com/guide/platform/j8-jack.html

**Comment by frenchfrywpepper** [comment]
@danielamiao I investigated the PGP signing of our current jars, and it isn't actually happening. I think we should add it, but since this isn't a regression I'm going to go ahead and merge.

**Comment by danielamiao** [comment]
💯

**Pull request merged**